### ISSUE TITLE
Cannot upload dependency-check-report.xml

### ISSUE THREAD

**Issue by arradz** [SonarSecurityCommunity/dependency-check-sonar-plugin#140]
**Describe the bug**
I am running [OWASP Dependency check using Docker](https://hub.docker.com/r/owasp/dependency-check). It's just dockerized CLI.
My project is .NET Core 2.2 with React FrontEnd

I run dep check without any issues and get generated report files, but on the step where I sonar-scanner is uploading report I get some errors.

```bash
[2019-05-03T06:21:03.673Z] INFO: Sensor Dependency-Check [dependencycheck]
[2019-05-03T06:21:03.673Z] INFO: Process Dependency-Check report
[2019-05-03T06:21:03.673Z] WARNING: WARN: Analysis aborted due to: Mandatory elements are missing. Plugin is compatible to https://jeremylong.github.io/DependencyCheck/dependency-check.1.8.xsd, https://jeremylong.github.io/DependencyCheck/dependency-check.2.0.xsd
[2019-05-03T06:21:03.932Z] INFO: Upload Dependency-Check HTML-Report
[2019-05-03T06:21:03.932Z] INFO: Process Dependency-Check report (done) | time=429ms
[2019-05-03T06:21:03.932Z] INFO: Sensor Dependency-Check [dependencycheck] (done) | time=429ms
```
launching **sonar-scanner** with `/d:verbose=true` I get additional debug message shown
```bash
[2019-05-03T06:46:36.848Z] 06:46:36.659 DEBUG: Vulnerability - cvssScore not found
[2019-05-03T06:46:36.848Z] org.sonar.dependencycheck.parser.ReportParserException: Vulnerability - cvssScore not found
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.parser.ReportParser.lambda$processVulnerability$7(ReportParser.java:189)
[2019-05-03T06:46:36.848Z] 	at java.util.Optional.orElseThrow(Optional.java:290)
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.parser.ReportParser.processVulnerability(ReportParser.java:189)
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.parser.ReportParser.processVulnerabilities(ReportParser.java:144)
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.parser.ReportParser.processDependency(ReportParser.java:123)
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.parser.ReportParser.processDependencies(ReportParser.java:87)
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.parser.ReportParser.parse(ReportParser.java:76)
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.DependencyCheckSensor.parseAnalysis(DependencyCheckSensor.java:61)
[2019-05-03T06:46:36.848Z] 	at org.sonar.dependencycheck.DependencyCheckSensor.execute(DependencyCheckSensor.java:93)
[2019-05-03T06:46:36.848Z] 	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
[2019-05-03T06:46:36.848Z] 	at org.sonar.scanner.sensor.ProjectSensorsExecutor.execute(ProjectSensorsExecutor.java:49)
[2019-05-03T06:46:36.848Z] 	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:320)
[2019-05-03T06:46:36.848Z] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
[2019-05-03T06:46:36.848Z] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
[2019-05-03T06:46:36.848Z] 	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:128)
[2019-05-03T06:46:36.848Z] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
[2019-05-03T06:46:36.848Z] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
[2019-05-03T06:46:36.848Z] 	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:73)
[2019-05-03T06:46:36.848Z] 	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:67)
[2019-05-03T06:46:36.848Z] 	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
[2019-05-03T06:46:36.848Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2019-05-03T06:46:36.848Z] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2019-05-03T06:46:36.848Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2019-05-03T06:46:36.848Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2019-05-03T06:46:36.848Z] 	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
[2019-05-03T06:46:36.848Z] 	at com.sun.proxy.$Proxy0.execute(Unknown Source)
[2019-05-03T06:46:36.848Z] 	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:171)
[2019-05-03T06:46:36.848Z] 	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:128)
[2019-05-03T06:46:36.848Z] 	at org.sonarsource.scanner.cli.Main.execute(Main.java:111)
[2019-05-03T06:46:36.848Z] 	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
[2019-05-03T06:46:36.848Z] 	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
[2019-05-03T06:46:36.848Z] 
```

mine **dependency-check-report.xml** _header_ looks like this
```xml
<?xml version="1.0"?>
<analysis xmlns="https://jeremylong.github.io/DependencyCheck/dependency-check.2.0.xsd">
  <scanInfo>
    <engineVersion>5.0.0-M2</engineVersion>
...
```
I am running **dependency check with** the following **arguments**:
```bash
--scan /src
--project MyProject
--out /report
--format ALL
--disableNodeJS
```

I don't know if it's important, but my whole setup is docker in docker (DinD), I am performing my .NET Core sonar scan by launching `nosinovacao/dotnet-sonar:latest`, performing build, test and dependency check inside by launching another docker container, mentioned above.

**Current behavior**
I get the error mentioned above and Sonar only can show HTML output.

**Expected behavior**
dependency-check-report.xml is imported into Sonar project.

**Versions (please complete the following information):**
 - dependency-check - 4.0.2
 - sonarqube - Community EditionVersion 7.7 (build 23042)
 - dependency-check-sonar-plugin - 1.2.3

**Comment by Reamer** [comment]
Hi @arradz 
> React FrontEnd

React is NodeJS, maybe you hit the same bug as in issue #135
Can you verify this?

**Comment by thib3113** [comment]
@arradz just for informations : 
`--disableNodeJS`

Doesn't totaly disable NodeJs analyser .

DependencyCheck use 3 analyzers for node package ( node package => all dependencies stored in `package.json` or `package-lock.json` or `npm-shrinkwrap.json` ) : 

- the National Vulnerability Database: https://nvd.nist.gov
- NPM Public Advisories: https://www.npmjs.com/advisories ( command `npm audit` ) 
- [the RetireJS community](https://github.com/RetireJS/retire.js)

The command argument `--disableNodeJS` will only disable the NVD + npm audit, but not `retireJs` . ( it seems `--disableNodeJS` contain the argument `--disableNodeAudit` )

So, if you want to totally disable node / npm analysis add this : `--disableRetireJS`
Else, I think it can be related to the issue #135 ( because npm audit doesn't return a score, but only a string, so cvssScore is not found ) .

**Comment by amandel** [comment]
I face the same issue with a Java & Angular JS project. 

```
17:14:01  [DEBUG] 15:14:01.075 Depedency Node projectReferences is not used
17:14:01  [WARNING] 15:14:01.076 Analysis aborted due to: Mandatory elements are missing. Plugin is compatible to https://jeremylong.github.io/DependencyCheck/dependency-check.1.8.xsd, https://jeremylong.github.io/DependencyCheck/dependency-check.2.0.xsd
17:14:01  [DEBUG] 15:14:01.077 Vulnerability - cvssScore not found
```

Stacktrace is (mostly) the same as above. 

    dependency-check - 5.0.0-M3
    sonarqube - 6.7.6 (build 38781) 
    dependency-check-sonar-plugin - 1.1.4

In general I'd rather drop this one finding and possibly create a technical finding here, than ignore the whole report?

I'll try with `--disableRetireJS` and feedback.

**Comment by thib3113** [comment]
@amandel You use dependency-check V 5.0.0-M3, in this version, the node analyzer use the xsd V2.1 so : `Plugin is compatible to https://jeremylong.github.io/DependencyCheck/dependency-check.1.8.xsd, https://jeremylong.github.io/DependencyCheck/dependency-check.2.0.xsd` . The plugin is not compatible . ( but you can disable the js checker )

**Comment by amandel** [comment]
@thib3113 Thanks for the feedback. Actually I do not want to disable the checker but get it all going nicely. :) As expected also with `--disableRetireJS` I get the same message.
And as said above the stacktrace is also "cvssScore not found" - even with the xsd change in 5.0.0-M3. 

I've not much time to spent, but tell me if I can support here.

**Comment by thib3113** [comment]
@amandel :
adding : `--disableRetireJS` on the dependency-check cli generate an xml with npm dependencies ??? ( Or you have another dependency with vulnerability and no cvssScore ? Not coming from NPM ? ) 

If you want a quick fix, use `5.0.0-M2` ( the xsd used will be the 2.0 and not the 2.1 ) . 

In fact, the real problem that NPM vulnerability doesn't have a `score` but only a severity level ( so, not a number . But a string containing `low`, `medium` ... ), so dependency-check can't add a score in the xml, so dependecy-check-sonar-plugin can't read it .

**Comment by amandel** [comment]
@thib3113 sorry, I've to correct this. I simply set disableRetireJS to true but some other options seem to have overweritten this (likely aso setting the retireJsUrl). So loking in the log and the xml file directly I see that the `RetireJS Analyzer` was still running. The use of the 5.0.0-M2 does not fix this.

**Comment by glianeric** [comment]
I have the same problem with Java analysis.   It seems that some dependencies (eg. xalan 2.7.0) do not have a CVSSv3_BaseScore.
The error I get:
[DEBUG] 17:25:19.431 Vulnerability - cvssScore not found
org.sonar.dependencycheck.parser.ReportParserException: Vulnerability - cvssScore not found
	at org.sonar.dependencycheck.parser.ReportParser.lambda$processVulnerability$8(ReportParser.java:221)

**Comment by Reamer** [comment]
@glianeric Can you upload your xalan 2.7.0 xml report or parts of it? Please use https://gist.github.com/
Following things are also needed
 - version of dependency-check?

**Comment by glianeric** [comment]
I am using version 5.0.0-M3.
I also ran a schema verification vs the 2.1 XSD and my results are valid.   Unfortunately the analysis is against internal components - I will run this against an OSS project in a moment and see if it causes the same problems.  If so, I will provide that XML.

**Comment by Reamer** [comment]
Can you check the [Snapshot-Build](https://github.com/SonarSecurityCommunity/dependency-check-sonar-plugin/releases/tag/1.2.4-SNAPSHOT) if this issue is still present?

**Comment by glianeric** [comment]
It appears to be working now. :)

**Comment by Reamer** [comment]
Thanks for the feedback, I'll prepare a release.

**Comment by Reamer** [comment]
Fixed with [1.2.4](https://github.com/SonarSecurityCommunity/dependency-check-sonar-plugin/releases/tag/1.2.4)
